docs/streams/developer-guide/processor-api.html

```diff
@@ -396,12 +400,68 @@ <h2>
                 </ul>
                 </p>
             </div>
+            <div class="section" id="versioned-state-stores">
+                <span id="streams-developer-guide-state-store-versioned"></span><h3><a class="toc-backref" href="#id12">Versioned Key-Value State Stores</a><a class="headerlink" href="#versioned-state-stores" title="Permalink to this headline"></a></h3>
+                <p>Versioned key-value state stores are available since Kafka Streams 3.5.
+                    Rather than storing a single record version (value and timestamp) per key,
+                    versioned state stores may store multiple record versions per key. This
+                    allows versioned state stores to support timestamped retrieval operations
+                    to return the latest record (per key) as of a specified timestamp, which
+                    enables use cases such as temporal stream-table joins.</p>
```

Comment:
`which enables use cases such as temporal stream-table joins.` -- I would remove this -- we have s-t-join already, so "enables" sounds weird. -- We also don't need to "justify/explain" why we have a store in the docs.